src/App.tsx

```diff
@@ -1,8 +1,81 @@
 import React from 'react';
 import './App.css';
+import { Goods } from './Goods';
+import { getGoods } from './api/api';
 
-const App = () => (
-  <h1>Dynamic list of Goods</h1>
-);
+interface AppState {
+  isLoading: boolean;
+  goods: Good[];
+}
 
-export default App;
+export class App extends React.Component<{}, AppState> {
+  state = {
+    isLoading: false,
+    goods: [],
+  };
+
+  loadAllGoods = () => {
+    this.setState({
+      isLoading: true,
+    });
+
+    getGoods()
+      .then(goods => {
+        this.setState({ goods, isLoading: false });
+      });
+  };
+
+  loadFirstFiveGoods = () => {
+    this.setState({
+      isLoading: true,
+    });
+
+    getGoods()
+      .then(goods => {
+        this.setState({
+          goods: goods
+            .sort((a: Good, b: Good) => a.name.localeCompare(b.name))
+            .filter((good: Good) => good.id <= 5),
```

Comment:
good: Good - getGoods() - specify returtn type instead

Reply:
Если ты сделал типизацию того что отдает функция getGoods, то это уже не нужно
```suggestion
            .sort((a, b) => a.name.localeCompare(b.name))
            .filter((good) => good.id <= 5),
```